In [109]:
import pandas

## Reading xlsx file

In [110]:
data=pandas.read_excel('./traningData.xlsx',header=None)

In [111]:
data

0     1
0   2.81  5.62
1   7.14  8.00
2   2.72  5.44
3   3.87  7.74
4   1.90  3.80
5   7.82  8.00
6   7.02  8.00
7   5.50  8.00
8   9.15  8.00
9   4.87  8.00
10  8.08  8.00
11  5.58  8.00
12  9.13  8.00
13  0.14  0.28
14  2.00  4.00
15  5.47  8.00
16  0.80  1.60
17  4.37  8.00
18  5.31  8.00
19  0.00  0.00
20  1.78  3.56
21  3.45  6.90
22  6.13  8.00
23  3.53  7.06
24  4.61  8.00
25  1.76  3.52
26  6.39  8.00
27  0.02  0.04
28  9.69  8.00
29  5.33  8.00
..   ...   ...
70  5.39  8.00
71  3.47  6.94
72  6.48  8.00
73  4.11  8.00
74  1.85  3.70
75  8.79  8.00
76  0.13  0.26
77  1.44  2.88
78  5.96  8.00
79  3.42  6.84
80  1.89  3.78
81  1.98  3.96
82  5.26  8.00
83  0.39  0.78
84  6.05  8.00
85  1.99  3.98
86  1.58  3.16
87  3.99  7.98
88  4.35  8.00
89  6.71  8.00
90  2.58  5.16
91  7.37  8.00
92  5.77  8.00
93  3.97  7.94
94  3.65  7.30
95  4.38  8.00
96  8.06  8.00
97  8.05  8.00
98  1.10  2.20
99  6.65  8.00

[100 rows x 2 columns]

## After looking at datas i am finding that once battery is full charging time doesn't affect the amonut of time battery lasts.hence i am finding the min charing time after which the battery is full

In [112]:
min=0
for index,row in data.iterrows():
    if(min==0 and row[1]==8):
        min =row[0]
    elif(min!=0 and row[1]==8):
        if(row[0]<min):
            min=row[0]
print min

4.11


## After iterating the dataFrame i have found that the min time required to recharge the battery completely is 4.11 unit

## hence i am considering only those rows which has x value lesser than minimum required time

In [113]:
newdata=data[data[1]!=8]

## index reset

In [114]:
newdata=newdata.reset_index(drop=True)

In [115]:
newdata

0     1
0   2.81  5.62
1   2.72  5.44
2   3.87  7.74
3   1.90  3.80
4   0.14  0.28
5   2.00  4.00
6   0.80  1.60
7   0.00  0.00
8   1.78  3.56
9   3.45  6.90
10  3.53  7.06
11  1.76  3.52
12  0.02  0.04
13  2.06  4.12
14  2.24  4.48
15  1.11  2.22
16  2.33  4.66
17  1.83  3.66
18  1.14  2.28
19  1.75  3.50
20  2.16  4.32
21  1.26  2.52
22  3.97  7.94
23  1.98  3.96
24  1.66  3.32
25  2.04  4.08
26  3.77  7.54
27  1.83  3.66
28  2.15  4.30
29  1.58  3.16
30  1.27  2.54
31  3.47  6.94
32  1.85  3.70
33  0.13  0.26
34  1.44  2.88
35  3.42  6.84
36  1.89  3.78
37  1.98  3.96
38  0.39  0.78
39  1.99  3.98
40  1.58  3.16
41  3.99  7.98
42  2.58  5.16
43  3.97  7.94
44  3.65  7.30
45  1.10  2.20

In [116]:
from sklearn import datasets, linear_model
import numpy as np

## lets train the data using Linear regression 

In [117]:
regr = linear_model.LinearRegression()

## I could have used cross validation for selecting the best coefficients but it looks like the datas satisfy this linear eqn y=2x+c. so i am not getting much into it

In [133]:
train_X_data=newdata[0][:-10]
test_X_data=newdata[0][-10:] ## test_data

In [120]:
train_X_data=np.reshape(train_X_data,(len(train_X_data),1))
test_X_data=np.reshape(test_X_data,(len(test_X_data),1))

In [94]:
train_Y_data=newdata[1][:-10]
test_Y_data=newdata[1][-10:]

In [122]:
train_Y_data=np.reshape(train_Y_data,(len(train_Y_data),1))
test_Y_data=np.reshape(test_Y_data,(len(test_Y_data),1))

In [124]:
regr.fit(train_X_data,train_Y_data)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=1, normalize=False)

In [125]:
regr.predict(2)

array([[ 4.]])

In [129]:
def predict(x):
    if(x>=min):
        return 8
    elif(x<=0):
        return "invalid value"
    else:
        return regr.predict(x)
predict(3)

array([[ 6.]])

In [130]:
predict(-5)

'invalid value'

In [135]:
predict(1.5)

array([[ 3.]])